File: src/lib/gov-exam/algorithmCatalog.ts
/**
 * Client projection of shared/algorithm-catalog.json.
 * Preview / UX only — Edge + Python validators are authoritative.
 */
import catalog from "./algorithmCatalog.json";

export const ALGORITHM_CATALOG = catalog;

export const QUALITY_ALGORITHM_VERSION = catalog.quality_algorithm_version;
export const DEDUP_ALGORITHM_VERSION = catalog.dedup_algorithm_version;
export const MASTERY_ALGORITHM_VERSION_CATALOG = catalog.mastery_algorithm_version;
export const SCORING_ALGORITHM_VERSION = catalog.scoring_algorithm_version;
export const PAPER_BLUEPRINT_VERSION = catalog.paper_blueprint_version;
export const CREDIT_CATALOG_VERSION = catalog.credit_catalog_version;

export const MIN_BANK_QUESTION_QUALITY = catalog.quality.min_bank_question_quality;
export const QUALITY_WEIGHTS = catalog.quality.weights;
export const DEDUP_POLICY = catalog.dedup;

File: scraper/app/shared/algorithm_catalog.py
"""Load the shared algorithm catalog (quality + dedup versions/weights)."""
from __future__ import annotations

import json
from functools import lru_cache
from pathlib import Path
from typing import Any

_HERE = Path(__file__).resolve().parent
_CANDIDATES = (
    _HERE / "algorithm_catalog.json",
    _HERE.parents[2] / "shared" / "algorithm-catalog.json",
)


@lru_cache(maxsize=1)
def load_algorithm_catalog() -> dict[str, Any]:
    for path in _CANDIDATES:
        if path.is_file():
            return json.loads(path.read_text(encoding="utf-8"))
    raise FileNotFoundError("algorithm_catalog.json is missing from the Python package")


def quality_spec() -> dict[str, Any]:
    return load_algorithm_catalog()["quality"]


def dedup_spec() -> dict[str, Any]:
    return load_algorithm_catalog()["dedup"]


def quality_algorithm_version() -> str:
    return str(load_algorithm_catalog()["quality_algorithm_version"])


def dedup_algorithm_version() -> str:
    return str(load_algorithm_catalog()["dedup_algorithm_version"])


def paper_blueprint_version() -> str:
    return str(load_algorithm_catalog()["paper_blueprint_version"])
